sale_delivery_split_date/models/sale_order_line.py

```diff
@@ -21,3 +19,9 @@ def _get_procurement_group_key(self):
                 # group by date instead of datetime
                 return (priority, self.commitment_date.date())
         return key
+
+    def _prepare_procurement_values(self, group_id=False):
+        vals = super(SaleOrderLine, self)._prepare_procurement_values(group_id)
+        if self.commitment_date:
+            vals.update({"date_planned": self.commitment_date})
```

Comment:
I see in the previous method that the priority is checked before using `commitment_date`, why aren't we checking priority here too?

Also, why is this new method needed in `14.0`?